src/main/java/com/thindeck/dynamo/DyTxn.java

```diff
@@ -41,28 +43,42 @@
  * @author Piotr Kotlicki (Piotr.Kotlicki@gmail.com)
  * @version $Id$
  */
-@Immutable
 public final class DyTxn implements Txn {
+
     /**
      * Table name.
      */
     public static final String TBL = "txns";
+
     /**
      * Transaction attribute.
      */
     public static final String ATTR_ID = "id";
+
+    /**
+     * Steps taken from scenario.
+     */
+    private final transient Iterator<Step> steps;
```

Comment:
@yegor256 Iterator is used to keep the state of iteration, if we were to use `com.jcabi.immutable.Array`, we would have to add another field that will keep the position of next step (and this fields will not be final). So the class can not be immutable.